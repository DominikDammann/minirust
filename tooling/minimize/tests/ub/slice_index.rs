//! Ensures an out-of-bounds index operation without an inbounds-check is UB.

fn main() {
    let x = [1, 2, 3_u8];
    let y: &[u8] = &x[..2];
    // UB: the index is out-of-bounds even though it is still in the allocation of x.
    // This generates an assume condition, which is then caught by MiniRust
    let _z = unsafe { y.get_unchecked(2) };
}
