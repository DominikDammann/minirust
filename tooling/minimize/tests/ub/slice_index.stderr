fatal error: UB: `Assume` intrinsic called on condition that is violated
